recipe/meta.yaml

```diff
@@ -1,92 +1,115 @@
-  sha256: 4804320e1e6cbe3a5421997b52199e3c1a3829b2ecb6489641da4b8e32faf500
+  fn: {{ vtag }}.tar.gz
+  url: https://github.com/oneapi-src/oneTBB/archive/{{ vtag }}.tar.gz
+  sha256: b182c73caaaabc44ddc5ad13113aca7e453af73c1690e4061f71dfe4935d74e8
+  patches:
+    - 0001-CMake-fix-Clang-compiler-detection-on-macOS.patch  # [osx]
 
 build:
-  number: 3
-  script_env:
-    - OS    # [win]
+  number: 0
 
 requirements:
   build:
     - python *
     - {{ compiler('cxx') }}
-    - m2w64-make                                         # [win]
-    # macOS hangs with conda's make for some reason
-    - make                                               # [linux]
+    - ninja  # [not osx]
+    - make   # [osx]  # CMake inserts -MD key during the Ninja build that causes compilation errors
+    - cmake >=3.13
 
 test:
   requires:
     - python *               # any python version is ok for sake of testing of shared libraries
   commands:
-    - python -c "import ctypes; assert {{ vinterface }} == ctypes.cdll[r'libtbb.so.2']       ['TBB_runtime_interface_version']()"  # [linux]
-    - python -c "import ctypes; assert {{ vinterface }} == ctypes.cdll[r'libtbb${SHLIB_EXT}']['TBB_runtime_interface_version']()"  # [unix and not linux]
-    - python -c "import ctypes, os; os.add_dll_directory(os.environ['LIBRARY_BIN']); assert {{ vinterface }} == ctypes.cdll[r'tbb.dll'] ['TBB_runtime_interface_version']()"  # [win]
+    - python -c "import ctypes; assert {{ vinterface }} == ctypes.cdll[r'libtbb.so.{{ vbinary }}.{{ vminor }}']       ['TBB_runtime_interface_version']()"  # [linux]
+    - python -c "import ctypes; assert {{ vinterface }} == ctypes.cdll[r'libtbb.{{ vbinary }}.{{ vminor }}${SHLIB_EXT}']['TBB_runtime_interface_version']()"  # [unix and not linux]
+    - python -c "import ctypes, os; os.add_dll_directory(os.environ['LIBRARY_BIN']); assert {{ vinterface }} == ctypes.cdll[r'tbb{{ vbinary }}.dll'] ['TBB_runtime_interface_version']()"  # [win]
 
 outputs:
   - name: tbb
     build:
-      script: {{ build }} --install-libs
+      script:
+        - set CMAKE_GENERATOR=Ninja  # [not osx]
+        - cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX={{ install_prefix }} -DTBB_TEST=OFF -S . -B build
+        - cmake --build build --parallel
+        - cmake -DCOMPONENT=runtime -P build/cmake_install.cmake
 
   - name: tbb-devel
     build:
-      script: {{ build }} --no-rebuild --install-devel --install-docs
+      script:
+        - set CMAKE_GENERATOR=Ninja  # [not osx]
+        - cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX={{ install_prefix }} -DTBB_TEST=OFF -S . -B build
+        - cmake --build build --parallel
+        - cmake -DCOMPONENT=devel -P build/cmake_install.cmake
       run_exports:
-        # excellent compatibility: https://abi-laboratory.pro/index.php?view=timeline&l=tbb
-        - {{ pin_subpackage('tbb', max_pin=None) }}
+        - {{ pin_subpackage('tbb', min_pin='2021') }}
```

Comment:
I don't fully understand the meaning of this line, but changed it since oneTBB 2021 breaks compatibility with previous versions. @anton-malakhov, could you please take a look?